src/neo/SmartContract/Helper.cs

```diff
@@ -146,6 +146,9 @@ internal static bool VerifyWitnesses(this IVerifiable verifiable, StoreView snap
             if (hashes.Length != verifiable.Witnesses.Length) return false;
             for (int i = 0; i < hashes.Length; i++)
             {
+                if (i != 0) gas -= verifiable.Witnesses[i - 1].GasConsumed;
```

Comment:
The last witness's gas is not subtracted, Imagine only one witness, this condition never happend

Reply:
This is OK, as we don't need to subtract in this case.

Reply:
Why? you are splitting the validation in two steps.
Imagine that have two witness, the first without state(A), the second with state (B). In this order[ A, B ] and do you have 1000 of gas.

- First you verify A because it's stateless, and compute `GasConsumed` 600 of GAS.
- You skip the B verification.
- Now you have the state verification.
- You skip A, because it's the firs position and it's state less.
- Second you verify B, 600 of GAS, you doesn't sum `A`, so you have gas with only 1000.

Reply:
Please take a look to https://github.com/neo-project/neo/pull/1507/commits/0ab3d71fa363f82665b4ed59e2d8c5af90b53840

Reply:
Got it